src/som/interpreter/objectstorage/ClassFactory.java

```diff
@@ -80,6 +92,14 @@ public ClassFactory(final SSymbol name, final MixinDefinition mixinDef,
 
     this.superclassAndMixins = superclassAndMixins;
 
+    if (VmSettings.SNAPSHOTS_ENABLED) {
+      CompilerDirectives.transferToInterpreter();
```

Comment:
This `transferToInterpreter()` shouldn't be done here, because it interferes with the  `VM.callerNeedsToBeOptimized` a few lines below. Perhaps initializing `serializationRoot` should come afterwards. Either way, we expect this code to be executed only on the slow path.